vaadin-portlet/src/main/java/com/vaadin/flow/portal/PortletWebComponentBootstrapHandler.java

```diff
@@ -141,11 +139,18 @@ private boolean checkWebpackConnection() {
                     .getAttribute(DevModeHandler.class.getName());
         }
         try {
-            DevModeHandler.getDevModeHandler().prepareConnection("/", "GET")
-                    .getResponseCode();
-            VaadinPortlet.getCurrent().getPortletContext()
-                    .setAttribute(DevModeHandler.class.getName(), true);
-            return true;
+            VaadinService vaadinService = VaadinPortletService.getCurrent();
+            Optional<DevModeHandler> devModeHandler =
+                    DevModeHandlerManager.getDevModeHandler(vaadinService);
+            if (devModeHandler.isPresent()) {
+                devModeHandler.get().prepareConnection("/", "GET")
+                        .getResponseCode();
+                VaadinPortlet.getCurrent().getPortletContext()
+                        .setAttribute(DevModeHandler.class.getName(), true);
+                return true;
+            } else {
+                return false;
```

Comment:
Should this only set false or should it go through and add the attribute also

Reply:
Very good question! Bootstrap handler code prepares the connection only once and doesn't try again if fails. The previous implementation was setting `false` for the attribute, i.e. no extra connection attempts would be made and I didn't notice any problems with it so far. Let's keep the behaviour as it was.

Reply:
https://github.com/vaadin/flow/pull/13622